### ISSUE TITLE
Using --metadata-file to set metadata does not seem to work properly

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#613]
Opening this issue to discuss how `--metadata-files` is supposed to work. 

This short example is working 
````markdown
---
author: "me"
format: html
metadata-files: 
  - meta.yml
---

# Hello

World 
````

where `meta.yml` is 
````yaml
title: My Document
````

I would have expected the equivalent command line to work too if the file was not passed through yaml
````markdown
---
author: "me"
format: html
---

# Hello

World 
````

````sh
quarto render test.qmd --metadata-file=meta.yml
````

But in that case the title is not set. The metadata does not see 

I look through the code @dragonstyle and have some question - maybe you can help.
https://github.com/quarto-dev/quarto-cli/blob/1fd5db82b8d757ff9425247dc3660d037ff63322/src/command/render/flags.ts#L286-L294

`readYamlFromMarkdownFile` returns a promise I believe after calling `readAndValidateYamlFromMarkdown`
https://github.com/quarto-dev/quarto-cli/blob/1fd5db82b8d757ff9425247dc3660d037ff63322/src/core/yaml.ts#L179-L185

and `readAndValidateYamlFromMarkdown` seems to expect a YAML block with `---` surrounding. I don't think `--metadata-file` is expecting that with Pandoc. 

When `metadata-files` is uses in YAML, the function used is `readAndValidateYamlFromFile` 
https://github.com/quarto-dev/quarto-cli/blob/1fd5db82b8d757ff9425247dc3660d037ff63322/src/config/metadata.ts#L60-L70
which is a different function so the two way of reading the same file - it does not seem right 🤔 

Anyway, I'll keep looking, but wanted to share so that maybe you can help @dragonstyle - am I missing something ? 

thanks!

**Comment by cscheid** [comment]
@jjallaire I've addressed this on e7bb9d9915915e93e44cd2139e8b7dbbb15b51a8. Note that @cderv pointed out a discrepancy between the two behaviors, which I made uniform. Specifically, the CLI flag now expects a "pure" yaml file, while in the past it expected yaml to be extracted from a markdown file (ie. collections of `---`-enclosed yaml chunks). I'm flagging it here so that you can make sure this is ok.

**Comment by jjallaire** [comment]
That looks great, thanks! (made one small comment on the commit)
